src/otx/algo/detection/heads/base_head.py

```diff
@@ -55,8 +55,12 @@ def get_positive_infos(self) -> list[InstanceData] | None:
             positive_infos.append(pos_info)
         return positive_infos
 
-    def loss(self, x: tuple[Tensor], entity: DetBatchDataEntity) -> dict:
-        """Perform forward propagation and loss calculation of the detection head.
+    def get_preds_and_targets(
```

Comment:
Let's combine them. I agree

Reply:
Additionally, I think it would be better to combine common parts across forwards (loss, predict, export) if there is.
https://github.com/openvinotoolkit/training_extensions/pull/3860#discussion_r1728972851

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/3865/commits/2e392af37a8f4f677dcfce5665844f6c8d96ac25
Predict and export can be combined in next PR to not extend this one more